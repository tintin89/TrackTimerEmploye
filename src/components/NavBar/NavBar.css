.navBar{
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: #1c1f24;
    height: 5rem;   
}
.navBar a{
  text-decoration: none;
}
.log{
  font-family: 'Poppins', sans-serif;
  color: #ffff;
  cursor: pointer;
  margin-right: 2rem;
}
.logo__container{
    display: flex;
    align-items: center;    
    cursor: pointer;
    
}
.logo__container span{
    color: #fff;
    
    display: flex;
    flex-wrap: wrap;
    max-width: 120px;
    font-family: 'Poppins', sans-serif;
    font-weight: 500;
}
.logo{
    height: 50px;
    pointer-events: none;
    animation: App-logo-spin infinite 10s linear;
}
@keyframes App-logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }