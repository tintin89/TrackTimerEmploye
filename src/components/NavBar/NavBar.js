import React from 'react';
import './NavBar.css';
import logo from '../../logo.svg';
import {useSelector,useDispatch} from 'react-redux';
import { logout } from '../../store/actions';
import axios from 'axios';
import moment from 'moment';
import {NavLink} from 'react-router-dom';


const mapState = (globalState)=>({
    userInfo:globalState.userInfo,
    arriving:globalState.arrivingHour
})

function NavBar() {
   
    const {userInfo,arriving} = useSelector(mapState);
    const dispatch = useDispatch();
    return (
        <div className="navBar">
            <NavLink to="/">
            <div className="logo__container">
            <img src={logo} alt="logo" className="logo"/>       
            <span>TRACK TIMER EMPLOYES</span>
            </div>
            </NavLink>
            {userInfo!=="" ? <span 
            onClick={async ()=>{
              if(JSON.parse(localStorage.getItem("isLunching"))===false || (!localStorage.getItem("isLunching"))){
                try {
                    await axios.patch(`https://timertracker-a17c6-default-rtdb.firebaseio.com/employers/${userInfo.localId}/${moment().format("MMM Do YY")}.json`,{
                        workedHours:JSON.parse(localStorage.getItem("employeTimer")).counter,
                        exitingHour:moment().format('LT'),
                        arrivingHour:arriving
                    })
                } catch (error) {
                    console.log(error);
                }
              
               return dispatch(logout());
              }else alert("you must finish the lunch!")
            }} 
            className="log">Log Out</span>
            :
            <NavLink to="/login">
                <span className="log">Log In</span>
            </NavLink>
        }     
        </div>
    )
}

export default NavBar
