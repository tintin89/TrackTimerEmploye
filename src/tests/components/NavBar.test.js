
import NavBar from '../../components/NavBar/NavBar'
import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('test in Employes.js',()=>{
    const store = mockStore({
        userinfo:"",
        errorM:"",
        loading:true
    });
    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
              <NavBar/>  
            </MemoryRouter>
            
        </Provider> )

    test('Must show correctly',()=>{
     
        expect(wrapper).toMatchSnapshot();
    })
})