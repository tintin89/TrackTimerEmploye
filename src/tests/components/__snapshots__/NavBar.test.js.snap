// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in Employes.js Must show correctly 1`] = `
<div
  className="navBar"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="logo__container"
    >
      <img
        alt="logo"
        className="logo"
        src="logo.svg"
      />
      <span>
        TRACK TIMER EMPLOYES
      </span>
    </div>
  </a>
  <span
    className="log"
    onClick={[Function]}
  >
    Log Out
  </span>
</div>
`;
